src/fe-common/core/fe-ignore.c

```diff
@@ -157,7 +157,8 @@ static void cmd_ignore(const char *data)
 		g_strsplit(chanarg, ",", -1);
 
 	rec = ignore_find_full(servertag, mask, patternarg, channels,
-			  IGNORE_FIND_PATTERN | ((level & MSGLEVEL_NO_ACT) ? IGNORE_FIND_NOACT : 0));
+			  IGNORE_FIND_PATTERN | ((level & MSGLEVEL_NO_ACT) ? IGNORE_FIND_NOACT : 0) |
```

Comment:
this is getting too ugly, we should add a temp variable flags= IGNORE_FIND_PATTERN and ...if (level &...) flags |= ...;